registration/manager.go

```diff
@@ -88,9 +88,28 @@ func NewSelfRegistrationManager(conf *SelfRegistrationManagerConfig, client clie
 	}
 }
 
-// SelfRegisterService registers Consul-Terraform-Sync with Consul
-func (m *SelfRegistrationManager) SelfRegisterService(ctx context.Context) error {
+// Start starts the self-registration manager, which will register CTS as a service
+// with Consul and deregister it if CTS is stopped.
+func (m *SelfRegistrationManager) Start(ctx context.Context) error {
+	// Register CTS with Consul
+	err := m.register(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Wait until the context is cancelled, initiate deregistration
+	<-ctx.Done()
+	err = m.deregister(ctx)
+	if err != nil {
+		return err
+	}
+	return ctx.Err()
+}
+
+// register registers Consul-Terraform-Sync with Consul
+func (m *SelfRegistrationManager) register(ctx context.Context) error {
```

Comment:
@lornasong I renamed this to be simply `register()` from `SelfRegisterService()` to simplify it and keep parity with `deregister()`, but left all the logs that still clarify that it's a self-registration of a service. Additionally, I was thinking that now that it's a `SelfRegistrationManager`, then the `Self` part was taken care of. Open to discussion, though, since I know you had some concerns earlier!

Reply:
I like `register()`! I think it removes redundancy as you're saying